src/file.rs

```diff
@@ -113,8 +114,8 @@ where
         stream.into()
     }
 
-    fn add_headers(&self, h: &mut header::Headers) {
-        h.set(header::ContentType(self.inner.content_type.clone()));
+    fn add_headers(&self, h: &mut http::header::HeaderMap) {
+        h.extend(self.inner.headers.clone());
```

Comment:
Can we avoid cloning the map itself? This is ugly but seems to work:

```
h.extend(self.inner.headers.iter().map(|(k, v)| (k.clone(), v.clone())));
```

Reply:
Fixed

Reply:
A `RefCell<Option<HeaderMap>>` might also work. Would you prefer that?

Reply:
I'd rather not do the `RefCell` thing. Currently `ChunkedReadFile` is `Clone`, and I'd like to preserve that (say, for putting in a cache to avoid having to open/stat/close repeatedly) without cache hits being subtly wrong (missing headers).